nomos-services/cryptarchia-consensus/src/storage/adapters/storage.rs

```diff
@@ -0,0 +1,78 @@
+// STD
+use std::hash::Hash;
+use std::marker::PhantomData;
+// Crates
+use nomos_core::block::Block;
+use nomos_core::header::HeaderId;
+use nomos_mempool::backend::MemPool;
+use nomos_storage::backends::StorageBackend;
+use nomos_storage::{StorageMsg, StorageService};
+use overwatch_rs::services::relay::OutboundRelay;
+use overwatch_rs::services::ServiceData;
+use serde::de::DeserializeOwned;
+// Internal
+use crate::storage::StorageAdapter as StorageAdapterTrait;
+
+pub struct StorageAdapter<Storage, ClPool, DaPool>
+where
+    Storage: StorageBackend + Send + Sync,
+{
+    pub storage_relay: OutboundRelay<<StorageService<Storage> as ServiceData>::Message>,
+    _cl_pool: PhantomData<ClPool>,
+    _da_pool: PhantomData<DaPool>,
+}
+
+#[async_trait::async_trait]
+impl<Storage, ClPool, DaPool> StorageAdapterTrait for StorageAdapter<Storage, ClPool, DaPool>
+where
+    Storage: StorageBackend + Send + Sync,
+    DaPool: MemPool<BlockId = HeaderId, Item: Clone + Eq + Hash + Send + DeserializeOwned> + Sync,
```

Comment:
You can separate this here:

```suggestion
    DaPool: MemPool<BlockId = HeaderId> + Sync,
    DaPool::Item:  Clone + Eq + Hash + Send + DeserializeOwned
```